src/components/graph.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from "react";
+import {
+  LineChart,
+  Label,
+  Line,
+  ResponsiveContainer,
+  XAxis,
+  YAxis,
+} from "recharts";
+
+export interface GraphData {
+  data: { x: number | string; y: number }[];
+  xLabel?: string;
+  yLabel?: string;
+}
+
+export function Graph(props: GraphData) {
+  return (
+    <ResponsiveContainer width="80%" aspect={2} className="sf-mx-auto sf-mt-2">
+      <LineChart data={props.data}>
+        <XAxis dataKey="x">
+          <Label
+            value={props.xLabel || ""}
+            offset={-5}
+            position="insideBottom"
+          />
+        </XAxis>
+        <YAxis allowDecimals={false}>
+          <Label
+            value={props.yLabel || ""}
+            angle={-90}
+            style={{ textAnchor: "middle" }}
+            position="insideLeft"
+          />
+        </YAxis>
+        <Line dataKey="y" />
+      </LineChart>
+    </ResponsiveContainer>
+  );
+}
+
+const possibleXlabels = [
+  "Time",
+  "Date",
+  "Event",
+  "Product",
+  "Category",
+  "Grades",
+  "Days",
+  "Months",
+  "Hours",
+  "Teams",
+  "State",
+  "City",
+  "Country",
+  "Models",
+  "Versions",
+  "Classes",
+  "Years",
+  "Quarters",
+  "Genre",
+  "Groups",
+  "Label",
+];
+
+const possibleYlabels = [
+  "Value",
+  "Price",
+  "Weight",
+  "Size",
+  "Length",
+  "Width",
+  "Height",
+  "Distance",
+  "Quantity",
+  "Score",
+  "Population",
+  "Revenue",
+  "Sales",
+  "GDP",
+  "Exposure",
+  "Temperature",
+  "Speed",
+  "Frequency",
+  "Density",
+  "UnitsSold",
+  "Satisfaction",
+];
+
+export function extractGraphData(data: string): GraphData | null {
+  console.log("extracting graph data", data);
```

Comment:
Sadly, we don't want big logs like this in `chat-ui`, since this will run in users' applications, so will fill up _their_ browser consoles

Reply:
Side point: once we start implementing in others' codebases, we'll want to somehow implement some kind of telemetry when things break so we can see bugs.

I think we can use sentry for this still, but not 100% sure about that

Reply:
yeah this is another log i forgot to remove